corehq/apps/translations/utils.py

```diff
@@ -25,12 +25,11 @@ def update_app_translations_from_trans_dict(app, trans_dict):
         app_translation_dict = app.translations
 
     if toggles.PARTIAL_UI_TRANSLATIONS.enabled(app.domain):
-        for lang, trans in six.iteritems(app.langs):
-            if lang in trans_dict:
-                if lang in app_translation_dict:
-                    app_translation_dict[lang].update(trans_dict[lang])
-                else:
-                    app_translation_dict[lang] = trans_dict[lang]
+        for lang in trans_dict:
```

Comment:
Couldn't `trans_dict` be missing a language that was just added to `app.langs`? What prevents that?

Reply:
trans_dict would be languages that need to be updated, so if it's missing a newly added lang that means users don't want to update that?

also related now, when user removes language, we would never remove its translations from app?

Reply:
> when user removes language, we would never remove its translations from app?

Skimming [edit_app_langs](https://github.com/dimagi/commcare-hq/blob/11c7845e122157ab2408b21da6ee6ca0f4934931/corehq/apps/app_manager/views/apps.py#L655), I don't see anything that alters `app.translations`, so no, I don't think we ever remove translations, though it doesn't look difficult to add. It also doesn't look like we add new languages to `app.translations`, which is why I think this code needs to handle them.

Reply:
> which is why I think this code needs to handle them.

My understanding is that `trans_dict` is generated from the excel sheet uploaded. So it would have all the languages the user wants to update (all in case of default upload and few in case of partial upload). 

Once we download the bulk *commcare* translation sheet, it has all the `app.langs`, so once user would upload it back it would have all the languages and hence in the `trans_dict`.

Are you saying we should be adding an empty dict here for the language that is in `app.langs` but missing in `trans_dict`?
If yes, i don't think thats needed because of what i said above.

Apology if this is something really simple and i am misunderstanding it. Just trying to confirm what i understand.

Reply:
You're right, thank you for the thorough explanation :)  I was confusing `trans_dict` and `app_translation_dict`...and on that note, what would you think of renaming `trans_dict`/`app_translation_dict` to something like `new_translations`/`old_translations` or `new_trans_dict`/`old_trans_dict`?

Reply:
ya, was considering renaming them. will do.

Reply:
Updated, https://github.com/dimagi/commcare-hq/pull/24548/commits/579d8109de663552ac31265b94f289d46916fbf7